TaskServiceInterview.cs

```diff
@@ -0,0 +1,448 @@
+            {
+                task.AssigneeInitials = null;
+                task.Status = TaskStatus.New;
+            }
+            else
+            {
+                task.AssigneeInitials = taskForward.AssigneeInitials;
+            }
+
+            task.SetClassificationAndRecalculatePermission(this.classificationRepository.Get(taskForward.ClassificationId));
+            task.IsHighPriority = taskForward.IsHighPriority;
+
+            // clear data requested by user
+            task.LockedByInitials = null;
+            task.LockedByTimestamp = null;
+
+            this.taskValidator.Validate(task, TaskValidationOperations.Forward);
+
+            var lockEventTime = this.GetLockTime(task.Id, employeeInitials, now);
+
+            var forwardBusinessEventData = new ForwardBusinessEventData
+            {
+                NewClassificationId = taskForward.ClassificationId.ToString(),
+                OldClassificationId = oldClassification.Id.ToString(),
+                NewOwner = taskForward.AssigneeInitials,
+                OldOwner = old,
+                TimeToCompleteTicks = (now - lockEventTime).Ticks,
+                DoneAfterDeadline = task.Deadline < now
+            };
+
+            task.SetActionExecutorInformation(ChangeReason.Forward, employeeInitials, this.clock.Current);
+            task.AddBusinessEventForward(employeeInitials, now, forwardBusinessEventData);
+
+            this.auditLogServant.Log("Task forwarded", new { taskForward, employeeInitials });
+            this.taskChangeNotificationServant.NotifyClassificationChanged(task, oldClassification);
+
+            this.notificationService.NotifyTaskOwnerAboutTaskAssignment(task, employeeInitials);
+
+            return true;
+        }
+
+        private TaskDetailsItem MapToTaskDetails(TaskEntity task)
+        {
+            var taskDetails = this.mappingServant.Map<TaskEntity, TaskDetailsItem>(task);
+            taskDetails.LockedByInitials = this.taskLockServant.IsLocked(task) ? task.LockedByInitials : null;
+            return taskDetails;
+        }
+
+
+        private IEnumerable<TaskListItem> ConvertToTaskListItems(IEnumerable<TaskEntity> tasks)
+        {
+            foreach (var task in tasks)
+            {
+                var taskListItem = this.mappingServant.Map<TaskEntity, TaskListItem>(task);
+                taskListItem.InternalReminderCounter = this.internalReminderFinder.GetNumberOfTaskInternalReminders(task.Id);
+                taskListItem.LockedByInitials = this.taskLockServant.IsLocked(task) ? task.LockedByInitials : null;
+                yield return taskListItem;
+            }
+        }
+
+        private void CheckPermissionsAndSkills(Guid taskId, string userInitials)
+        {
+            if (!this.permissionServant.CanPerformAction(taskId, userInitials))
+                throw new InsufficientPrivilegesException();
+        }
+
+        private void CheckIfCanFinishOrArchive(Guid taskId, string userInitials)
```

Comment:
This method is private and is not used anywhere. Following the YAGNI principle, we should not implement functionality until it is necessary.